zcash_client_backend/src/data_api/chain.rs

```diff
@@ -140,6 +140,8 @@ pub trait BlockSource {
 /// - `block_source` Source of compact blocks
 /// - `from_tip` Height & hash of last validated block; if no validation has previously
 ///    been performed, this will begin scanning from `sapling_activation_height - 1`
+/// - `limit` amount of blocks that will be fetched from `block_source` and attempt to 
+///    be validated
```

Comment:
```suggestion
/// - `limit`: Maximum number of blocks that should be validated in this call.
```

The documentation also needs to explain that if `limit` is provided, then the caller is responsible for repeatedly calling this function until no more validation occurs. It's not immediately clear to me whether the caller can do this however, or if the return value of this function needs to be altered to return the `(height, hash)` of the last validated block (so that the caller can use this on the next call, and terminate once the returned tuple equals `from`).

Reply:
I'm hesitating on how to solve this tuple as return value. Mainly because when limit provided is `None`, there's a fallback on sapling activation height, which is just a height and its has is an Optional.

```
 let from_height = validate_from
        .map(|(height, _)| height)
        .unwrap_or(sapling_activation_height - 1);
```

Reply:
The safest thing to do here is to query the wallet for the latest scanned height each time. For that reason, I've changed this back to returning `Result<(), ...>`.